### PR TITLE
Change docs to use Pydata theme

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

This is the theme used by numpy, scipy, pandas, etc and looks much better than our current one.
Also did a little bit of reformatting, started a user guide, etc, but there is more to be done

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by tinosulzer** [comment]
Docs look like this on latest build: https://pybamm--2618.org.readthedocs.build/en/2618/
Don't know why the pydata theme isn't being used

**Comment by tinosulzer** [comment]
Almost ready to go, final problem is the code is not displaying on this page https://pybamm--2618.org.readthedocs.build/en/2618/source/user_guide/fundamentals/index.html (it's not displaying locally either)

**Comment by tinosulzer** [comment]
Found the problem, I was doing `.. code-block::python` instead of `.. code-block:: python` 🤦

File: docs/source/api/plotting/index.rst [hunk 1]
```diff
@@ -4,7 +4,6 @@ Plotting
 .. toctree::
 
   quick_plot
-  dynamic_plot
```
Comment (Saransh-cpp) [inline docs/source/api/plotting/index.rst:7]: Was this intentional?

**Review by Saransh-cpp — APPROVED** [review]
The documentation looks amazing! Thanks, @tinosulzer!

**Review by brosaplanella — APPROVED** [review]
Looks great, thanks! We should aim to merge this by Tuesday so we can have the new docs with the new release.

File: docs/source/api/plotting/index.rst (hunk 1 above)
Comment (tinosulzer) [reply docs/source/api/plotting/index.rst:7]: yes, the contents of that file are now in `quick_plot` since they go together

**Pull request merged**